pkg/ext-proc/backend/pod_reconciler.go

```diff
@@ -0,0 +1,73 @@
+package backend
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/gateway-api-inference-extension/api/v1alpha1"
+	logutil "sigs.k8s.io/gateway-api-inference-extension/pkg/ext-proc/util/logging"
+)
+
+type PodReconciler struct {
+	client.Client
+	Datastore *K8sDatastore
+	Scheme    *runtime.Scheme
+	Record    record.EventRecorder
+}
+
+func (c *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	inferencePool, err := c.Datastore.getInferencePool()
```

Comment:
we also need to handle delete events, if we get an even for a pod that doesn't exist or has its deletionTimestamp set and its namespace matches that of the inferencePool, we need to delete from the store (no need to check for the selector)